client/securedrop_client/config.py

```diff
@@ -2,29 +2,47 @@
 import logging
 import os
 
-logger = logging.getLogger(__name__)
-
-
-class Config:
-    CONFIG_NAME = "config.json"
+try:
+    from qubesdb import QubesDB
+except ImportError:
+    QubesDB = False
 
-    def __init__(self, journalist_key_fingerprint: str) -> None:
-        self.journalist_key_fingerprint = journalist_key_fingerprint
 
-    @classmethod
-    def from_home_dir(cls, sdc_home: str) -> "Config":
-        full_path = os.path.join(sdc_home, Config.CONFIG_NAME)
+logger = logging.getLogger(__name__)
 
-        try:
-            with open(full_path) as f:
-                json_config = json.loads(f.read())
-        except Exception as e:
-            logger.error("Error opening config file at {}: {}".format(full_path, e))
-            json_config = {}
 
-        return Config(
-            journalist_key_fingerprint=json_config.get("journalist_key_fingerprint", None)
-        )
+class Config:
+    """Configuration loaded at runtime from QubesDB (if available) or
+    environment variables."""
+
+    # Mapping of `Config` attributes (keys) to how to look them up (values)
+    # from either QubesDB or the environment.
+    mapping = {
+        "gpg_domain": "QUBES_GPG_DOMAIN",
```

Comment:
nit: Since you've put these side by side, should we take the opportunity to standardize the names a bit more, e.g. having them all start with "SD_"?

Reply:
[`QUBES_GPG_DOMAIN`](https://github.com/QubesOS/qubes-doc/blob/72052be4d5884e912152403ec6d15ca78bf0b1bf/user/security-in-qubes/split-gpg.md?plain=1#L102) comes from Qubes itself, not from us.